BEIMA.Client/src/pages/Devices/AddDevicePage.js

```diff
@@ -26,10 +28,32 @@ const AddDevicePage = () => {
   const [deviceImage, setDeviceImage] = useState();
   const [deviceAdditionalDocs, setAdditionalDocs] = useState();
   const [fullDeviceJSON, setFullDeviceJSON] = useState({});
+  const [deviceTypes, setDeviceTypes] = useState([]);
   
   useEffect(() => {
-    setPageName('Add Device')
-  })
+    setPageName('Add Device');
+    const loadData = async () => {
+      let types = await getDeviceTypes();
+      setDeviceTypes(types);
+    }
+   loadData()
+  },[])
+  
+  const getDeviceTypes = async () => {
+    const deviceTypeData = await GetDeviceTypeList();
+    let data = []
+    for(let i = 0; i < deviceTypeData.response.length; i++){
+      data.push({
+        name: deviceTypeData.response[i].name,
+        id: deviceTypeData.response[i].id
+      });
+    }
+    return data
+  }
+  
+  const getFieldsForTypeId = (deviceTypeId) => {
```

Comment:
My guess is that this will be completed in the PR for #168? Just wanted to check

Reply:
either that one or another specifically targeted task for the work. at some point I'll edit that task to be a specific one like #220 (the task for this PR) is.